jmemmgr.c

```diff
@@ -95,6 +95,15 @@ round_up_pow2(size_t a, size_t b)
 #endif
 #endif
 
+#ifdef WITH_SIMD
+#if (defined(__ARM_NEON) || defined(__ARM_NEON__)) && (ALIGN_SIZE % 32)
+  /* 32-byte alignment allows us to extract more performance from */
+  /* fancy-upsampling algorithms when using Neon. */
+  #error "Neon optimizations rely on 32-byte alignment. \
+          ALIGN_SIZE is not a multiple of 32 bytes."
+#endif
+#endif
+
```

Comment:
@jonathanwright-ARM `ALIGN_SIZE` is always 32 if `WITH_SIMD` is defined, so I'm not sure why this code is necessary.  Please comment.

Reply:
I read the comment starting on line 70 as saying that users can define ALIGN_SIZE themselves if they think their alignment needs are less than 32 bytes. This is technically true for, e.g. Arm systems with Neon enabled, and they may set ALIGN_SIZE = 16 thinking they'll save a few bytes here and there.
However, I have written the Neon optimizations for fancy-upsampling such that the assumed minimum output buffer size is 64 bytes (i.e. 2*ALIGN_SIZE). I added this extra compile-time check to make sure the alignment is what I think it is (the default here), else the code will likely scribble on memory it doesn't own.

Reply:
That is true for the other SIMD extensions as well.  There is no way for anyone to change `ALIGN_SIZE` without modifying the libjpeg-turbo source code, and as soon as they do that, they're on their own as far as making it work.

Reply:
It was extra caution on my part to make sure it doesn't fail silently if someone tries to be clever - if it's not warranted, discard the check.

Reply:
Yeah, the ability to override build settings in **jconfig.h** is a legacy of the pre-autotools days when libjpeg was introduced.  No one realistically does that anymore.